deepsensor/model/model.py

```diff
@@ -648,6 +648,597 @@ def unnormalise_pred_array(arr, **kwargs):
+                            else:
+                                prev_patch_x1_min = get_index(
+                                    patch_prev[var_name].coords[orig_x1_name].min(),
+                                    x1=True,
+                                )
+
+                                b_x1_min = (
+                                    prev_patch_x1_min - patch_x1_index[0]
+                                ) - patch_overlap[0]
+                        else:
+                            b_x1_max = b_x1_max + 1
+
+                        patch_clip_x1_min = int(b_x1_min)
+                        patch_clip_x1_max = int(
+                            data_array.sizes[orig_x1_name] - b_x1_max
+                        )
+                        patch_clip_x2_min = int(b_x2_min)
+                        patch_clip_x2_max = int(
+                            data_array.sizes[orig_x2_name] - b_x2_max
+                        )
+
+                        patch_clip = data_array.isel(
+                            **{
+                                orig_x1_name: slice(
+                                    patch_clip_x1_min, patch_clip_x1_max
+                                ),
+                                orig_x2_name: slice(
+                                    patch_clip_x2_min, patch_clip_x2_max
+                                ),
+                            }
+                        )
+
+                        patches_clipped[var_name].append(patch_clip)
+
+            combined = {
+                var_name: xr.combine_by_coords(patches, compat="no_conflicts")
+                for var_name, patches in patches_clipped.items()
+            }
+
+            return combined
+
+        # load patch_size and stride from task
+        patch_size = tasks[0]["patch_size"]
+        stride = tasks[0]["stride"]
+
+        # sanitise patch_size and stride arguments
+        if isinstance(patch_size, float) and patch_size is not None:
+            patch_size = (patch_size, patch_size)
+
+        if isinstance(stride, float) and stride is not None:
+            stride = (stride, stride)
+
+        if stride[0] > patch_size[0] or stride[1] > patch_size[1]:
+            raise ValueError(
+                f"stride must be smaller than patch_size in the corresponding dimensions for patchwise prediction. Got: patch_size: {patch_size}, stride: {stride}"
+            )
+
+        # patchwise prediction does not yet support more than a single date
+        num_task_dates = len(set([t["time"] for t in tasks]))
+        if num_task_dates > 1:
+            raise NotImplementedError(
+                f"Patchwise prediction does not yet support more than a single date at a time, got {num_task_dates}. \n\
+                                      Contributions to the DeepSensor package are very welcome. \n\
+                                      Please see the contributing guide at https://alan-turing-institute.github.io/deepsensor/community/contributing.html"
```

Comment:
I don't think this is necessary to mention here.

Reply:
[afac690](https://github.com/alan-turing-institute/deepsensor/pull/135/commits/afac690bd52e39dd6b57b9865da7f64698107963)